crates/compiler/test_syntax/tests/snapshots/pass/return_empty_assign.expr.formatted.roc

```diff
@@ -0,0 +1,3 @@
+return #
+    s
```

Comment:
This seems to lose the record restructure?

Reply:
That's intentional (and was the behavior even prior to this PR). What's changing in this PR is that now we're not accidentally missing a case we should have been doing it in.

The intent was that early in the development of `!` suffixes,users previously may have typed `{} = foo! 1 2 3`, in order to get around the requirement that all values are used. There was an effort to auto-upgrade any such usages via the formatter, since the assignment to the empty record was now implicit/fine with the next generation of puritty inference.

I'd be on board with re-opening the discussion of whether this "unit assignment removal" is actually pulling its weight (but again, separate discussion and separate PR I think).